### PR TITLE
diplomacy: a type of connect that always disables monitors

### PR DESCRIPTION
**Pull request by hcook**

### CODE REVIEW

File: src/main/scala/diplomacy/Nodes.scala [hunk 1]
```diff
@@ -18,7 +18,8 @@ trait InwardNodeImp[DI, UI, EI, BI <: Data]
   def bundleI(ei: EI): BI
   def colour: String
   def reverse: Boolean = false
-  def connect(edges: () => Seq[EI], bundles: () => Seq[(BI, BI)])(implicit p: Parameters, sourceInfo: SourceInfo): (Option[MonitorBase], () => Unit) = {
+  def connect(edges: () =>  Seq[EI], bundles: () => Seq[(BI, BI)], enableMonitoring: Boolean)
```
Comment (colinschmidt) [inline src/main/scala/diplomacy/Nodes.scala]: Not that it maters but you put an extra space before Seq[EI]

File: src/main/scala/rocket/Frontend.scala [hunk 2]
```diff
@@ -59,7 +59,7 @@ class Frontend(val icacheParams: ICacheParams, hartid: Int)(implicit p: Paramete
   val masterNode = TLOutputNode()
   val slaveNode = TLInputNode()
 
-  icache.slaveNode.map { _ := slaveNode }
+  icache.slaveNode.map { _ connectButDontMonitor slaveNode }
```
Comment (colinschmidt) [inline src/main/scala/rocket/Frontend.scala]: Why is this one no longer being monitored? The other place monitors are disabled has a nice comment.

File: src/main/scala/rocketchip/Periphery.scala [hunk 3]
```diff
@@ -349,8 +349,7 @@ trait HasPeripheryErrorSlave extends HasSystemNetworks {
 
   // Override the default Parameters to exclude the TLMonitor between the Fragmenter and error slave.
```
Comment (colinschmidt) [inline src/main/scala/rocketchip/Periphery.scala]: This comment could be updated now that its not an "override" of parameters.

"Most slaves...than others, therefore we exclude the TLMonitor."

**Comment by hcook** [comment]
i wanted to make `:/!/=" but @terpstra said i couldnt :(

**Pull request merged**